dbms/src/Storages/DeltaMerge/Remote/Serializer.cpp

```diff
@@ -112,6 +115,40 @@ RemotePb::RemoteSegment Serializer::serializeTo(
     return remote;
 }
 
+#ifdef DBMS_PUBLIC_GTEST
+static std::tuple<String, UInt64> parseDMFilePath(const String & path)
+{
+    // Path likes /disk1/data/t_100/stable/dmf_2.
+    auto pos = path.find_last_of('_');
+    RUNTIME_CHECK(pos != std::string::npos, path);
+    auto file_id = stoul(path.substr(pos + 1));
+
+    pos = path.rfind("/dmf_");
+    RUNTIME_CHECK(pos != std::string::npos, path);
+    auto parent_path = path.substr(0, pos);
+    return std::tuple<String, UInt64>{parent_path, file_id};
+}
+#endif
+
+static DMFilePtr restoreDMFile([[maybe_unused]] const Remote::IDataStorePtr & data_store, const String & remote_fname)
+{
+#ifndef DBMS_PUBLIC_GTEST
+    auto prepared = data_store->prepareDMFileByKey(remote_fname);
+    auto dmfile = prepared->restore(DMFile::ReadMetaMode::all());
+#else
+    auto [parent_path, file_id] = parseDMFilePath(remote_fname);
+    auto key_manager = std::make_shared<MockKeyManager>(false);
+    auto file_provider = std::make_shared<FileProvider>(key_manager, false);
+    auto dmfile = DMFile::restore(
+        file_provider,
+        file_id,
+        /*page_id*/ 0,
+        parent_path,
+        DMFile::ReadMetaMode::all());
+#endif
+    return dmfile;
+}
```

Comment:
I think this is too hacky. Maybe write a `class MockDataStore` and handle the testing logic there.
Then we can make the data_store as one param of `Serializer::deserializeSegmentSnapshotFrom` to write the testing logic.